src/rdata.cpp

```diff
@@ -818,7 +819,15 @@ void ReturnData::fFIM(int it, Model &model, const ExpData &edata) {
                     *
                     amici::fsres(y_it.at(iy), sy_it.at(iy + ny * jp),
                                  observedData[iy], sigmay_it.at(iy),
-                                 ssigmay_it.at(iy + ny * jp));
+                                 ssigmay_it.at(iy + ny * jp))
+                    -
+                    ssigmay_it.at(iy + ny * ip) * ssigmay_it.at(iy + ny * jp)
```

Comment:
Note that this term is already accounted for in `https://github.com/AMICI-dev/AMICI/blob/a4324d02c9b1c47c770b48f8427360fe61d5a1ff/src/rdata.cpp#L729`, but the sign may be wrong?

Reply:
Ah, okay, good to konw! Then the only term which we needed is the $\sigma^4$ thing.
And yes, the sign was wrong...
So, the FIM I think we want is basically:
(sy / sigma)^T * (sy / sigma) + (ssigma / sigma)^T * (ssigma / sigma) - ((data - y) *ssigmy / sigma^2)^T * ((data - y)  *ssigmy / sigma^2)^T